features/steps/errors.py

```diff
@@ -152,6 +143,46 @@ class InvalidValueError(RuleState):
     def __str__(self):
         return f"On instance {misc.fmt(self.inst)} the following invalid value for {self.attribute} has been found: {self.value}"
 
+@dataclass
+class InvalidPropertySetDefinition(RuleState):
+    inst: ifcopenshell.entity_instance
+    object: ifcopenshell.entity_instance
+    name: str = None
+    types: list = None
+    template_type_enum: str = None
```

Comment:
This is a bit contradictory. When 'name' is not given, it default to None. But this is not a string, so this will lead to an AttributeError. 
A way around it is Typing.Optional[str] = None. 

I've modified it in https://github.com/buildingSMART/ifc-gherkin-rules/pull/102/commits/b5d11133e964d69e66f9b35c419b2d95ba8686c5 . Hope that's ok

Reply:
Hi, good point. Thank you very much <3